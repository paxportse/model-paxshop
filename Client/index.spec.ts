import "isomorphic-fetch"
import * as dotenv from "dotenv"
import * as model from "../index"

dotenv.config()

describe("model.Client", () => {
	it("booking.fetch", async () => {
		const client = model.Client.create(
			process.env.apiUrl || "https://api.shop.paxport.group",
			"https://app.dummytravel.com/test"
		)
		expect(await client.booking.fetch({ reference: "AAAA12" })).toMatchSnapshot("booking-AAAA12")
	})
})
