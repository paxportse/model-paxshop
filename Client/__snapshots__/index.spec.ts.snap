// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`model.Client booking.fetch: booking-AAAA12 1`] = `
Object {
  "options": Object {
    "departure": Array [
      Object {
        "arrival": "2022-08-20T16:34:00.000Z",
        "departure": "2022-08-20T13:37:00.000Z",
        "from": "ARN",
        "meals": Array [
          Object {
            "alternatives": Array [
              Object {
                "description": "Some chocolate and so on.",
                "name": "Sparkling wine & Chocolate",
                "price": Object {
                  "amount": 200,
                  "currency": "SEK",
                },
              },
              Object {
                "description": "Some champagne and so on.",
                "name": "Champagne & Chocolate",
                "price": Object {
                  "amount": 300,
                  "currency": "SEK",
                },
              },
            ],
            "name": "Celebration",
          },
        ],
        "rows": Array [
          Object {
            "groups": Array [
              Object {
                "seats": Array [
                  Object {
                    "class": "first-class",
                    "legroom": true,
                    "price": Object {
                      "amount": 400,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
              Object {
                "seats": Array [
                  Object {
                    "class": "first-class",
                    "legroom": true,
                    "price": Object {
                      "amount": 400,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "first-class",
                    "legroom": true,
                    "price": Object {
                      "amount": 400,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
            ],
          },
          Object {
            "groups": Array [
              Object {
                "seats": Array [
                  Object {
                    "class": "first-class",
                    "legroom": true,
                    "price": Object {
                      "amount": 400,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
              Object {
                "seats": Array [
                  Object {
                    "class": "first-class",
                    "legroom": true,
                    "price": Object {
                      "amount": 400,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "first-class",
                    "legroom": true,
                    "price": Object {
                      "amount": 400,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
            ],
          },
          Object {
            "groups": Array [
              Object {
                "seats": Array [
                  Object {
                    "class": "first-class",
                    "legroom": true,
                    "price": Object {
                      "amount": 400,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
              Object {
                "toilet": true,
              },
            ],
          },
          Object {},
          Object {
            "groups": Array [
              Object {
                "seats": Array [
                  Object {
                    "class": "premium-economy",
                    "price": Object {
                      "amount": 300,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "premium-economy",
                    "price": Object {
                      "amount": 300,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
              Object {
                "seats": Array [
                  Object {
                    "class": "premium-economy",
                    "price": Object {
                      "amount": 300,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "premium-economy",
                    "price": Object {
                      "amount": 300,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
            ],
          },
          Object {
            "groups": Array [
              Object {
                "seats": Array [
                  Object {
                    "class": "premium-economy",
                    "price": Object {
                      "amount": 300,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "premium-economy",
                    "price": Object {
                      "amount": 300,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
              Object {
                "seats": Array [
                  Object {
                    "class": "premium-economy",
                    "price": Object {
                      "amount": 300,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "premium-economy",
                    "price": Object {
                      "amount": 300,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
            ],
          },
          Object {
            "groups": Array [
              Object {
                "seats": Array [
                  Object {
                    "class": "premium-economy",
                    "price": Object {
                      "amount": 300,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "premium-economy",
                    "price": Object {
                      "amount": 300,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
              Object {
                "seats": Array [
                  Object {
                    "class": "premium-economy",
                    "price": Object {
                      "amount": 300,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "premium-economy",
                    "price": Object {
                      "amount": 300,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
            ],
          },
          Object {
            "groups": Array [
              Object {
                "seats": Array [
                  Object {
                    "class": "premium-economy",
                    "price": Object {
                      "amount": 300,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "premium-economy",
                    "price": Object {
                      "amount": 300,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
              Object {
                "seats": Array [
                  Object {
                    "class": "premium-economy",
                    "price": Object {
                      "amount": 300,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "premium-economy",
                    "price": Object {
                      "amount": 300,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
            ],
          },
          Object {
            "groups": Array [
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
            ],
          },
          Object {
            "groups": Array [
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
            ],
          },
          Object {
            "groups": Array [
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
            ],
          },
          Object {
            "groups": Array [
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
            ],
          },
          Object {
            "exit": true,
            "groups": Array [
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "legroom": true,
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "legroom": true,
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "legroom": true,
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "legroom": true,
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
            ],
          },
          Object {
            "groups": Array [
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
            ],
          },
          Object {
            "groups": Array [
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
            ],
          },
          Object {
            "groups": Array [
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
            ],
          },
          Object {
            "groups": Array [
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
            ],
          },
          Object {
            "groups": Array [
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
            ],
          },
          Object {
            "groups": Array [
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
              Object {
                "seats": Array [
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                  Object {
                    "class": "economy",
                    "price": Object {
                      "amount": 200,
                      "currency": "SEK",
                    },
                    "status": "available",
                  },
                ],
              },
            ],
          },
        ],
        "to": "LHR",
      },
    ],
    "luggage": Array [],
  },
  "passengers": Array [
    Object {
      "ageGroup": "child",
      "name": Object {
        "first": "Pelle",
        "last": "Karlsson",
      },
    },
  ],
  "reference": "AAAA12",
}
`;
